File: python/LeetCode_8_StringtoInteger.py
def myAtoi(s: str) -> int:
    num = 0
    sign = 0
    is_negative = 0
    for letter in s:
        code = ord(letter)
        if (code >= 48 and code <= 57):
            num *= 10
            num += (code - 48)
            sign = 1
        elif (code == 45):
            if (sign): break
            sign = 1
            is_negative = 1
        elif (code == 43):
            if (sign): break
            sign = 1
        elif (code == 32):
            if (sign): break
            continue 
        else:
            break

    if (is_negative):
        num *= -1

    if (num <= -2147483648 or num >= 2147483647):
        return -2147483648 if is_negative else 2147483647
    return num


print(myAtoi("    -88827   5655  U"))

File: c/LeetCode_8_StringtoInteger.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>


int myAtoi(char* s) {
    int num = 0;
    int sign = 0;
    int is_negative = 0;
    for (int i = 0; i < strlen(s); ++i) {
        if (s[i] >= 48 && s[i] <= 57) {
            num *= 10;
            num += (s[i] - 48);
            sign = 1;
        } else if (s[i] == 45) {
            if (sign) break;
            sign = 1;
            is_negative = 1;
        } else if (s[i] == 43) {
            if (sign) break;
            sign = 1;
        } else if (s[i] == 32) {
            if (sign) break;
            continue;
        } else {
            break;
        }
    }

    if (is_negative) {
        num *= -1;
    }

    if (num <= -2147483648 || num >= 2147483647) {
        return (is_negative) ? -2147483648 : 2147483647;
    }

    return num;
}


int main() {
    char *s = "42";

    printf("%d ", myAtoi(s));
    return 0;
}